Sources/MongoSwift/ChangeStream.swift

```diff
@@ -281,4 +283,18 @@ public class ChangeStream<T: Codable>: CursorProtocol {
             self.wrappedCursor.kill()
         }
     }
+
+#if compiler(>=5.5) && canImport(_Concurrency) && os(Linux)
+    /// When concurrency is available, we can ensure change streams are always cleaned up properly.
+    deinit {
```

Comment:
I think this is a lot simpler than what I had to do for `ClientSession`s because the internal `Cursor` class already neatly packages up the state and stores a references to the `Connection`. 

`kill()` is defined here: https://github.com/mongodb/mongo-swift-driver/blob/f22e1fd3340982f58f0b87718d9651aa9dd76a43/Sources/MongoSwift/CursorCommon.swift#L354

And calls through to `close()`, where only once destroying the cursor/change stream completes does `self.state` get mutated and the reference dropped: https://github.com/mongodb/mongo-swift-driver/blob/f22e1fd3340982f58f0b87718d9651aa9dd76a43/Sources/MongoSwift/CursorCommon.swift#L219

And since we are calling `kill` on the `Cursor` itself here, the `Cursor` should stay alive until the method completes.

that said, would appreciate if y'all review the logic in that file to check my reasoning and confirm what I have here should work.

Reply:
Yeah I think this looks good. If this were future based I'd be a bit worried that `client` might go out of scope before `kill` completed, but I think the `await` here ensures it stays alive until the future is evaluated.